### PR TITLE
Python(feat): pytest graceful handle missing connection

### PR DESCRIPTION
**Pull request by alexluck-sift**
## Why

Target handled missing connection via `--sift-test-results-check-connection`, which yielded `None` from `step` / `report_context` / `module_substep` on ping failure. Any test calling `step.measure(...)` then crashed with `AttributeError`, forcing `if step is None: pytest.skip(...)` guards throughout user code. The flag name also didn't convey what it actually did.

## Changes

- Adds `--sift-offline`: skips the ping, routes create/update calls to a JSONL log file for replay via `import-test-result-log`, tolerates missing `SIFT_*` env vars.
- Adds `--sift-disabled` (also `SIFT_DISABLED=1`): nothing contacts Sift, no log file is written. `step.measure(...)` still evaluates bounds and returns a real pass/fail boolean. No credentials required.
- Exposes `is_simulated` on `TestReport` / `TestStep` / `TestMeasurement` / `ReportContext` so callers can branch on provenance (true for disabled and offline modes).
- Replaces the opaque check-connection toggle with two named modes that say what they do — `offline` (write to log) vs `disabled` (skip entirely).
- Online mode now fail-fasts: a failed ping aborts the session with `pytest.UsageError` naming both flags as escape hatches.
- Renames CLI/ini args for brevity (`sift_test_results_*` → `sift_*`).

### CODE REVIEW

File: python/lib/sift_client/_tests/conftest.py [hunk 1]
```diff
@@ -79,5 +79,16 @@ def ci_pytest_tag(sift_client):
...
+
+    Integration runs (``-m integration``) stay online with the log file
+    disabled so writes go inline against the real backend; missing
+    ``SIFT_*`` env vars surface as an actionable usage error from the
+    plugin's ``sift_client`` fixture. Every other run defaults to
+    ``--sift-disabled`` so unit tests don't need credentials.
+    """
+    is_integration_run = "integration" in (config.option.markexpr or "")
+    if is_integration_run:
+        config.option.sift_log_file = False
```
Comment (alexluck-sift) [inline python/lib/sift_client/_tests/conftest.py]: use log file instead

File: python/lib/sift_client/pytest_plugin.py [hunk 2]
```diff
@@ -248,37 +283,60 @@ def _report_context_impl(
...
         test_case = base_name
-    log_file = _resolve_log_file(pytestconfig)
+    # When the client is in simulate mode (disabled), bypass the log-file
+    # pipeline — pinning a path would create a file the simulate path never
+    # writes to — and skip the import subprocess (no real RPCs to replay).
+    simulate = sift_client._simulate
+    log_file: str | Path | bool | None = False if simulate else _resolve_log_file(pytestconfig)
     git_metadata = _option_or_ini(pytestconfig, _GIT_METADATA)
     include_git_metadata = True if git_metadata is None else bool(git_metadata)
+    offline = False if simulate else _is_offline(pytestconfig)
```
Comment (ian-sift) [inline python/lib/sift_client/pytest_plugin.py]: Took me a sec to confirm that offline still simulates since we are now deriving a `simulate` from this plugin where we then define `offline` as `!simulate` but then ultimately both paths still call `simulate_<request>`. Idk if keeping the terminology as `disabled` and `offline` would be clearer. 

Fine to keep as is though because I could still follow after re-reading. Can resolve this comment  or address it and ping me

**Review by ian-sift — APPROVED** [review]

File: python/lib/sift_client/pytest_plugin.py (hunk 2 above)
Comment (alexluck-sift) [reply python/lib/sift_client/pytest_plugin.py]: agreed, renamed some of the args to make it explicit what is being done. Now we are explicitly setting log replay.

**Review by ian-sift — APPROVED** [review]
